packages/ra-core/src/controller/edit/useEditController.ts

```diff
@@ -151,7 +151,7 @@ export const useEditController = <
                 : '',
     });
 
-    const recordCached = { id, previousData: record };
+    const recordCached = { id, previousData: record, meta: mutationMeta };
```

Comment:
this should no longer be necessary with the new implementation of `useUpdate`, as `mutationMeta` is already passed to `update()` at line 244